[CI] BlockHashRandomizedTests test {forcePackedHash=false groups=2 maxValuesPerPosition=1 dups=0 allowedTypes=[Basic[elementType=LONG], Basic[elementType=INT]]} failing

**Build Scans:**
- [elasticsearch-periodic-platform-support #12164 / ubuntu-2404_platform-support-unix](https://gradle-enterprise.elastic.co/s/hir62em6iqqqi)
- [elasticsearch-periodic-platform-support #12164 / ubuntu-2404_platform-support-unix](https://gradle-enterprise.elastic.co/s/smdu6cozrr6ua)
- [elasticsearch-periodic-platform-support #12164 / rocky-9_platform-support-unix](https://gradle-enterprise.elastic.co/s/3chqqkmsnwd4i)
- [elasticsearch-periodic-platform-support #12164 / amazonlinux-2023_platform-support-aws](https://gradle-enterprise.elastic.co/s/ttqeufpa25bas)
- [elasticsearch-pull-request #140916 / part-3](https://gradle-enterprise.elastic.co/s/g6r3ilsbku7ia)
- [elasticsearch-pull-request #140916 / part-3](https://gradle-enterprise.elastic.co/s/sud4bfrbody5g)
- [elasticsearch-pull-request #140893 / part-3](https://gradle-enterprise.elastic.co/s/j2ndakjvotezg)
- [elasticsearch-pull-request #140891 / part-3](https://gradle-enterprise.elastic.co/s/ajpkvlqjxrx4k)
- [elasticsearch-pull-request #140893 / part-3](https://gradle-enterprise.elastic.co/s/sl56gsowfqqmq)
- [elasticsearch-pull-request #140891 / part-3](https://gradle-enterprise.elastic.co/s/eysy6xhfkuuh2)

**Reproduction Line:**
```
./gradlew ":x-pack:plugin:esql:compute:test" --tests "org.elasticsearch.compute.aggregation.blockhash.BlockHashRandomizedTests.test {forcePackedHash=false groups=2 maxValuesPerPosition=1 dups=0 allowedTypes=[Basic[elementType=BOOLEAN], Basic[elementType=INT], Basic[elementType=LONG], Basic[elementType=DOUBLE], Basic[elementType=NULL], Basic[elementType=BYTES_REF], Ordinals[dictionarySize=10]]}" -Dtests.seed=C1D59213A72B67E5 -Dtests.locale=doi-IN -Dtests.timezone=Australia/Currie -Druntime.java=26
```

**Applicable branches:**
main

**Reproduces locally?:**
N/A

**Failure History:**
[See dashboard](https://es-delivery-stats.elastic.dev/app/dashboards#/view/dcec9e60-72ac-11ee-8f39-55975ded9e63?_g=(refreshInterval:(pause:!t,value:60000),time:(from:now-7d%2Fd,to:now))&_a=(controlGroupState:(initialChildControlState:('0c0c9cb8-ccd2-45c6-9b13-96bac4abc542':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:task.keyword,order:0,selectedOptions:!(),title:'GradleTask',type:optionsListControl),'4e6ad9d6-6fdc-4fcc-bf1a-aa6ca79e0850':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:className.keyword,order:1,selectedOptions:!(org.elasticsearch.compute.aggregation.blockhash.BlockHashRandomizedTests),title:'Suite',type:optionsListControl),'144933da-5c1b-4257-a969-7f43455a7901':(dataViewId:fbbdc689-be23-4b3d-8057-aa402e9ed0c5,fieldName:name.keyword,order:2,selectedOptions:!('test%20%7BforcePackedHash%3Dfalse%20groups%3D2%20maxValuesPerPosition%3D1%20dups%3D0%20allowedTypes%3D%5BBasic%5BelementType%3DLONG%5D%2C%20Basic%5BelementType%3DINT%5D%5D%7D'),title:'Test',type:optionsListControl)))))

**Failure Message:**
```
java.lang.AssertionError: 
Expected: (a value less than or equal to <1000> or <10000>)
     but: was <4096>
```

**Issue Reasons:**
- [main] 2 consecutive failures in test test {forcePackedHash=false groups=2 maxValuesPerPosition=1 dups=0 allowedTypes=[Basic[elementType=LONG], Basic[elementType=INT]]}
- [main] 2 consecutive failures in step ubuntu-2404_platform-support-unix
- [main] 10 failures in test test {forcePackedHash=false groups=2 maxValuesPerPosition=1 dups=0 allowedTypes=[Basic[elementType=LONG], Basic[elementType=INT]]} (11.0% fail rate in 91 executions)
- [main] 2 failures in step ubuntu-2404_platform-support-unix (100.0% fail rate in 2 executions)
- [main] 6 failures in step part-3 (15.4% fail rate in 39 executions)
- [main] 3 failures in pipeline elasticsearch-pull-request (8.3% fail rate in 36 executions)

**Note:**
This issue was created using new test triage automation. Please report issues or feedback to es-delivery.

--- a/muted-tests.yml
+++ b/muted-tests.yml
@@ -146,15 +146,9 @@ tests:
 - class: org.elasticsearch.xpack.esql.qa.single_node.EsqlSpecForceStoredLoadingIT
   method: test {csv-spec:fuse.fuseWithRowLinearAndMultiValueGroupColumn}
   issue: https://github.com/elastic/elasticsearch/issues/147613
-- class: org.elasticsearch.compute.aggregation.blockhash.BlockHashRandomizedTests
-  method: test {forcePackedHash=false groups=2 maxValuesPerPosition=1 dups=0 allowedTypes=[Basic[elementType=LONG], Basic[elementType=INT]]}
-  issue: https://github.com/elastic/elasticsearch/issues/147654
 - class: org.elasticsearch.xpack.transform.integration.TransformCrossProjectIT
   method: testUpdateTransformWithProjectRouting
   issue: https://github.com/elastic/elasticsearch/issues/147655
-- class: org.elasticsearch.compute.aggregation.blockhash.BlockHashRandomizedTests
-  method: testWithCranky {forcePackedHash=false groups=2 maxValuesPerPosition=1 dups=0 allowedTypes=[Basic[elementType=LONG], Basic[elementType=INT]]}
-  issue: https://github.com/elastic/elasticsearch/issues/147732
 - class: org.elasticsearch.xpack.esql.expression.function.scalar.spatial.StSimplifyPreserveTopologyTests
   method: testEvaluateBlockWithNulls
   issue: https://github.com/elastic/elasticsearch/issues/147773

--- a/x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/aggregation/blockhash/LongIntAdaptiveBlockHash.java
+++ b/x-pack/plugin/esql/compute/src/main/java/org/elasticsearch/compute/aggregation/blockhash/LongIntAdaptiveBlockHash.java
@@ -81,6 +81,15 @@ public final class LongIntAdaptiveBlockHash extends AdaptiveBlockHash {
         return intBlock.asVector();
     }
 
+    // for testing
+    int effectiveEmitBatchSize() {
+        if (current instanceof LongIntVectorOnlyBlockHash) {
+            return vectorBatchSize;
+        } else {
+            return emitBatchSize;
+        }
+    }
+
     final class LongIntVectorOnlyBlockHash extends BlockHash {
         private final LongLongHashTable longLongHash;
         private final long batchUsedBytes;

--- a/x-pack/plugin/esql/compute/src/test/java/org/elasticsearch/compute/aggregation/blockhash/BlockHashRandomizedTests.java
+++ b/x-pack/plugin/esql/compute/src/test/java/org/elasticsearch/compute/aggregation/blockhash/BlockHashRandomizedTests.java
@@ -200,9 +200,13 @@ public class BlockHashRandomizedTests extends ComputeTestCase {
                          * page (the vector/vector fast path). Page sizes are assumed safe, so
                          * emitting exactly positionCount in one shot is acceptable.
                          */
+                        int effectiveEmitBatchSize = emitBatchSize;
+                        if (blockHash instanceof LongIntAdaptiveBlockHash adaptive) {
+                            effectiveEmitBatchSize = adaptive.effectiveEmitBatchSize();
+                        }
                         assertThat(
                             ordsAndKeys.ords().getTotalValueCount(),
-                            either(lessThanOrEqualTo(emitBatchSize)).or(equalTo(positionCount))
+                            either(lessThanOrEqualTo(effectiveEmitBatchSize)).or(equalTo(positionCount))
                         );
                     }
                     batchCount[0]++;

